### PR TITLE
Add additional test cases to scale down budget tests.

### PR DESCRIPTION
**Pull request by hbostan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Tests were missing cases where the empty and drain nodes are from the same atomic group.

#### Special notes for your reviewer:
This is a follow-up pr addressing this comment: https://github.com/kubernetes/autoscaler/pull/6034/files#r1291199619

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by hbostan** [comment]
cc @aleksandra-malinowska

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go [hunk 1]
```diff
@@ -298,6 +362,62 @@ func TestCropNodesToBudgets(t *testing.T) {
 			wantEmpty:                generateNodeGroupViewList(testNg, 0, 4),
 			wantDrain:                generateNodeGroupViewList(testNg, 0, 2),
 		},
+		"empty&drain atomic nodes with deletions in progress, overall budget exceeded, both empty&drain nodes fit": {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scaledown/budgets/budgets_test.go:365]: >both empty&drain nodes fit

nit: it looks like only empty nodes fit in this test case? Unless I'm misunderstanding something :)

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go [hunk 2]
```diff
@@ -298,6 +362,62 @@ func TestCropNodesToBudgets(t *testing.T) {
...
+		},
+		"empty&drain atomic nodes in same group with deletions in progress, drain budget exceeded, both empty&drain nodes fit": {
+			emptyDeletionsInProgress: 2,
+			drainDeletionsInProgress: 2,
+			empty:                    generateNodeGroupViewList(atomic5, 0, 1),
+			drain:                    generateNodeGroupViewList(atomic5, 1, 5),
+			wantEmpty:                generateNodeGroupViewList(atomic5, 0, 1),
+			wantDrain:                generateNodeGroupViewList(atomic5, 1, 5),
+		},
+		"empty&drain regular and atomic nodes with deletions in progress, overall budget exceeded, only atomic is drained": {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/scaledown/budgets/budgets_test.go:397]: >only atomic is drained

nit: it seems that in this test case nothing is drained, only the entirely empty atomic group is removed?

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go (hunk 1 above)
Comment (hbostan) [reply cluster-autoscaler/core/scaledown/budgets/budgets_test.go]: You are right, fixed it :)

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go (hunk 2 above)
Comment (hbostan) [reply cluster-autoscaler/core/scaledown/budgets/budgets_test.go]: Fixed the test case name

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go [hunk 3]
```diff
@@ -234,6 +259,24 @@ func TestCropNodesToBudgets(t *testing.T) {
...
 		},
+		"empty&drain regular and atomic nodes in same group exceeding overall limit, no deletions in progress": {
+			empty:     append(generateNodeGroupViewList(testNg, 0, 5), generateNodeGroupViewList(atomic11, 0, 8)...),
+			drain:     generateNodeGroupViewList(atomic11, 8, 11),
+			wantEmpty: generateNodeGroupViewList(atomic11, 0, 8),
+			wantDrain: generateNodeGroupViewList(atomic11, 8, 11),
+		},
+		"empty&drain regular and multiple atomic nodes in same group exceeding drain limit, no deletions in progress": {
+			empty:     append(append(generateNodeGroupViewList(testNg, 0, 5), generateNodeGroupViewList(atomic8, 0, 5)...), generateNodeGroupViewList(atomic11, 0, 8)...),
+			drain:     append(generateNodeGroupViewList(atomic11, 8, 3), generateNodeGroupViewList(atomic8, 5, 8)...),
```
Comment (BigDarkClown) [inline cluster-autoscaler/core/scaledown/budgets/budgets_test.go:270]: `generateNodeGroupViewList(atomic11, 8, 3)` - shouldn't `from < to`? Now this returns no nodes

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go [hunk 4]
```diff
@@ -234,6 +259,24 @@ func TestCropNodesToBudgets(t *testing.T) {
...
+			wantEmpty: append(generateNodeGroupViewList(atomic8, 0, 5), generateNodeGroupViewList(testNg, 0, 2)...),
+			wantDrain: generateNodeGroupViewList(atomic8, 5, 8),
+		},
+		"empty&drain multiple atomic nodes in same group exceeding overall limit, no deletions in progress": {
+			empty:     append(append(generateNodeGroupViewList(atomic3, 0, 3), generateNodeGroupViewList(atomic4, 0, 2)...), generateNodeGroupViewList(atomic11, 0, 8)...),
```
Comment (BigDarkClown) [inline cluster-autoscaler/core/scaledown/budgets/budgets_test.go:275]: `atomic11` has 8 nodes, I think it should have 11.

File: cluster-autoscaler/core/scaledown/budgets/budgets_test.go (hunk 3 above)
Comment (hbostan) [reply cluster-autoscaler/core/scaledown/budgets/budgets_test.go]: Nice catch! It should be correct now.

**Pull request merged**